test/e2e/storage/testsuites/volumeperf.go

```diff
@@ -0,0 +1,287 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testsuites
+
+import (
+	"context"
+	"fmt"
+	"github.com/davecgh/go-spew/spew"
+	"sync"
+	"time"
+
+	"github.com/onsi/ginkgo"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/watch"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epv "k8s.io/kubernetes/test/e2e/framework/pv"
+	"k8s.io/kubernetes/test/e2e/storage/testpatterns"
+)
+
+type volumePerformanceTestSuite struct {
+	tsInfo TestSuiteInfo
+}
+
+var _ TestSuite = &volumePerformanceTestSuite{}
+
+const (
+	provisionCount     = 500
+	testTimeout        = 15 * time.Minute
+	expectedLatency    = 5 * time.Minute
+	expectedThroughput = float64(1)
```

Comment:
I think these numbers should be configurable per storage driver because performance and supported number of volumes will vary considerably. We used arguments before, but that turned out to be not easily discoverable. Better extend https://github.com/kubernetes/kubernetes/blob/613ee7f0dbc148e088e67ee7b4f13d97f9655ee0/test/e2e/storage/testsuites/testdriver.go#L195-L202 and skip the test if those options are not set.

The volume size also should be configurable.

Reply:
Agree let's make this configurable and we could enable the test for hostpath and mock csi drivers

Reply:
Done